### PR TITLE
Update to DiscordJS v14, reformatting and update `/confess` command

### PR DESCRIPTION
**Pull request by NovAti0n**
This PR updates all the dependencies to their newer version, reformats all files using Prettier and updates the `/confess` command to use [`message components`](https://discord.com/developers/docs/interactions/message-components) instead of reactions.

Here are the whole list of modifications:

- Change the confess command to include buttons instead of reactions.
- Update `discord.js` to v14, `@discordjs/rest` to v2, `discord-api-types` to v0.37 and `dotenv` to v16. This was done to make the use of message components easier.
- Change the whole structure to remove `package.json` from the `src` directory (and change all paths accordingly).
- Remove unnecessary Dockerfile and Procfile.
- Format all files to follow `.prettierrc` rules.
- Update `README.md` to add instructions.
- Update `.gitignore` to include `package-lock.json`, as [this is considered a good practice](https://docs.npmjs.com/cli/v9/configuring-npm/package-lock-json).
- Remove the `/crash` command as it was deemed useless.

All commands have been tested for compatibility issues with the dependencies. Patches were applied if needed. I opened a few issues on bugs that are not related to the update and that were already present before. These should be fixed in other PRs.

As this PR modifies a lot of files, extra caution would be advisable when deploying the changes. The bot should be monitored for any abnormal behaviour after deployement.

### CODE REVIEW

**Comment by obiwac** [comment]
you're gonna pass me as biggest contributor 😪

**Review by Emilevillette — APPROVED** [review]
Looks good to me.

File: README.md [hunk 1]
```diff
@@ -2,31 +2,32 @@
...
 
 - [x] An anonymous confession system. Members send a DM to the bot (`/confess <message>`), the admin approve it in a
-  private channel (with a react) and the message will be then in the public channel)
+      private channel (with a react) and the message will be then in the public channel)
 - [x] A bulk clear (`/clear <number_message>`)
 - [ ] A meme contest event creator (`/createcontest <channel_id> <emote> <start_date> <end_date>`) with the score
-  
+
   command
+
```
Comment (obiwac) [inline README.md:30]: ?

File: src/commands/crash.js
Comment (obiwac) [inline src/commands/crash.js:1]: could we add back a similar command in the future which does what this is supposed to?

**Review by obiwac — CHANGES_REQUESTED** [review]

File: src/commands/crash.js
Comment (Emilevillette) [reply src/commands/crash.js:1]: @obiwac I have started that on the old repository. Looks like the branch hasn't been migrated tho

File: README.md (hunk 1 above)
Comment (obiwac) [reply README.md:30]: image

File: src/commands/crash.js
Comment (Emilevillette) [reply src/commands/crash.js:1]: No no, it was `meme-master`

Comment (obiwac) [reply src/commands/crash.js:1]: okay that's pushed too now

**Review by obiwac — APPROVED** [review]

**Pull request merged**